errors/error.go

```diff
@@ -33,6 +33,30 @@ func ListToError(errs []error) error {
 // Fields keeps context.
 type Fields map[string]interface{}
 
+func (f Fields) copy() Fields {
+	newFields := make(map[string]interface{})
```

Comment:
We can specifiy hashmap length since we know it initially 
```suggestion
	newFields := make(map[string]interface{}, len(f))
```